experimental/ubuntu/ganglia/Dockerfile

```diff
@@ -0,0 +1,99 @@
+  && tar xvjf $PHANTOM_JS.tar.bz2 \
+  && mv $PHANTOM_JS /usr/local/share \
+  && ln -sf /usr/local/share/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+# Apache2 config. The `sites-enabled` config files are loaded into the container
+# later.
+RUN rm /etc/apache2/sites-enabled/* && a2enmod proxy && a2enmod proxy_http
+# Install Ganglia API server & support for Ganglia Python modules
+# Remove all modules besides the nvidia.pyconf module (used for GPU metrics)
+# Note, we originally use "pip install -r ganglia-api/requirements.txt" to install ganglia-api's
+# dependency, which is tornado. However, this requirement file (https://github.com/guardian/ganglia-api/blob/master/requirements.txt)
+# does not specify the version and tornado will break PythonShell.py when it imports
+# from IPython.nbconvert.filters.ansi import ansi2html. It is probably caused by the fact that
+# IPython's version is too old. For now, we just pin the version of tornado to the version used by
+# Runtime 5.2. Once we move to Runtime 6, we should be able to use a newer version of tornado
+# because PythonShell should not depend on nbconvert.
+RUN export GANGLIA_API=ganglia-api-cbe773d051168e05118774708ff7a0ce881617f4 \
+  && wget https://s3-us-west-2.amazonaws.com/databricks-build-files/$GANGLIA_API.tar.gz \
+  && tar xvzf $GANGLIA_API.tar.gz \
+  && pip install tornado==5.1.1 \
+  && apt-get install ganglia-monitor-python \
+  && find /etc/ganglia/conf.d/*.pyconf ! -name 'nvidia.pyconf' -type f -exec rm -f {} + \
+  && find /etc/ganglia/conf.d/*.pyconf.disabled ! -name 'nvidia.pyconf' -type f -exec rm -f {} + \
+  && find /usr/lib/ganglia/python_modules/*.py ! -name 'nvidia.py' -type f -exec rm -f {} +
+# Stop Ganglia related services from starting on boot
+RUN update-rc.d ganglia-monitor disable \
+  && update-rc.d gmetad disable \
+  && update-rc.d apache2 disable
+
+# create directories for unix monitoring tool required for ganglia
+# two services needed by Ganglia are gmetad and ganglia-monitor
+RUN mkdir -p /etc/monit/conf.d
+
+SHELL ["/bin/bash", "-c"]
+
+RUN echo $'check process ganglia-monitor with pidfile /var/run/ganglia-monitor.pid\n\
+      start program = "/usr/sbin/service ganglia-monitor start"\n\
+      stop program = "/usr/sbin/service ganglia-monitor stop"\n\
+      if memory usage > 500 MB for 3 cycles then restart\n\
+' > /etc/monit/conf.d/ganglia-monitor-not-active
+
+RUN echo $'check process gmetad with pidfile /var/run/gmetad.pid\n\
+      start program = "/usr/sbin/service gmetad start"\n\
+      stop program = "/usr/sbin/service gmetad stop"\n\
+      if memory usage > 500 MB for 3 cycles then restart\n\
+\n\
+check process apache2 with pidfile /var/run/apache2/apache2.pid\n\
+      start program = "/usr/sbin/service apache2 start"\n\
+      stop program = "/usr/sbin/service apache2 stop"\n\
+      if memory usage > 500 MB for 3 cycles then restart\n\
+' > /etc/monit/conf.d/gmetad-not-active
```

Comment:
should be easier to do this via ADD or COPY docker primitives

Reply:
Updated to remove inline shell commands and used ADD